haas/api.py

```diff
@@ -258,34 +275,57 @@ def node_connect_network(node, nic, network):
 
     project = node.project
 
+    allocator = get_network_allocator()
+    if channel is None:
+        channel = allocator.get_default_channel(db)
+
```

Comment:
True. It was more about collecting the channel logic together so that it's easier to follow the channel logic flow.